app/models/user.rb

```diff
@@ -0,0 +1,17 @@
+class User < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable, :trackable,
+         :recoverable, :rememberable, :validatable, :omniauthable, omniauth_providers: [:github]
+
+  def self.from_omniauth(auth)
+    where(provider: auth.provider, uid: auth.uid).first_or_create do |user|
+      user.provider = auth.provider
+      user.name = auth.info.name
+      user.uid = auth.uid
+
+      user.email = auth.info.email
+      user.password = Devise.friendly_token[0,20]
```

Comment:
```suggestion
      user.password = Devise.friendly_token
```

The friendly token is by default 20 characters long, so no need to add that on top if the documentation is correct. 

Do you ever expect oauth users to also log in via password? Otherwise I'm not sure if default setting makes sense here.